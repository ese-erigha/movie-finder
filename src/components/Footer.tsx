import React from 'react';
import Image from 'react-bootstrap/Image';
import tmdbLogo from 'assets/img/tmdb_logo.png';

const Footer = () => (
  <footer className="footer text-center mt-4 mb-4 white">
    <span>&#169; 2021 made with</span>
    <i className="fa fa-heart heart space-left-7" aria-hidden="true" />
    <span className="space-left-7">by</span>
    <a href="https://github.com/eseerigha/movie-finder" className="space-left-7">
      eseerigha
    </a>
    <a
      data-testid="tmdb-link"
      href="https://developers.themoviedb.org/3/getting-started/introduction"
      rel="noopener noreferrer"
      target="_blank"
      className="space-left-15"
    >
      <Image data-testid="tmdb-logo" src={tmdbLogo} />
    </a>
  </footer>
);
export default Footer;
