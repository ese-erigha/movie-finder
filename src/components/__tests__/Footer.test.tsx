import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';

import Footer from 'components/Footer';

describe('Footer', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Footer />);
  });
  test('renders the component', () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  test('should contain github account', () => {
    expect(screen.getByText('eseerigha')).toBeInTheDocument();
  });
  test('should render tmdb section', () => {
    expect(screen.getByTestId('tmdb-logo')).toBeInTheDocument();
    expect(screen.getByTestId('tmdb-link')).toBeInTheDocument();
  });
  test('should contain footer text', () => {
    expect(screen.getByText(/2021 made with/)).toBeInTheDocument();
  });
});
