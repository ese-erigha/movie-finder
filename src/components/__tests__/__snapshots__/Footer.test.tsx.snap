// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the component 1`] = `
<DocumentFragment>
  <footer
    class="footer text-center mt-4 mb-4 white"
  >
    Â© 2021 made with
    <i
      aria-hidden="true"
      class="fa fa-heart heart space-left-7"
    />
    <span
      class="space-left-7"
    >
      by
    </span>
    <a
      class="space-left-7"
      href="https://github.com/eseerigha/movie-finder"
    >
      eseerigha
    </a>
    <a
      class="space-left-15"
      data-testid="tmdb-link"
      href="https://developers.themoviedb.org/3/getting-started/introduction"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        class=""
        data-testid="tmdb-logo"
        src="tmdb_logo.png"
      />
    </a>
  </footer>
</DocumentFragment>
`;
